components/log-viewer-webui/client/src/ui/MainLayout.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {useState} from "react";
+
+import {
+    SearchOutlined,
+    UploadOutlined,
+} from "@ant-design/icons";
+import {
+    Layout,
+    Menu,
+    MenuProps,
+} from "antd";
+
+import "./MainLayout.css";
+
+
+const {Sider} = Layout;
+
+type MenuItem = Required<MenuProps>["items"][number];
+
+const items: MenuItem[] = [
+    {label: "Ingest", key: "/ingest", icon: <UploadOutlined/>},
+    {label: "Search", key: "/search", icon: <SearchOutlined/>},
+];
+
+/**
+ * The main layout of web ui.
+ *
+ * @return
+ */
+const MainLayout: React.FC = () => {
```

Comment:
In tsx files, we intentionally omit the return types of functional components. See https://www.reddit.com/r/reactjs/comments/wjq51d/comment/ijivsrx/
```suggestion
const MainLayout = () => {
```